tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,395 @@
+      f.write(FOURSPACES+"Relative pixel spacing in X:1.00"+NEWLINE)
+      f.write(FOURSPACES+"Relative pixel spacing in Y:1.00"+NEWLINE)
+      f.write(FOURSPACES+"Relative pixel spacing in Z:1.00"+NEWLINE)  
+
+      for rule in nameandtypes["con_assign_a_name_to"]["r_match_rule"]:
+          f.write(FOURSPACES+"Select the rule criteria:"+ rule["match_all_any"] +" ("+rule["con_match"]["rule_type"]+" "+rule["con_match"]["operator"]+" "+rule["con_match"]["contain"]+" \""+rule["con_match"]["match_value"]+"\")"+NEWLINE)
+          f.write(FOURSPACES+"Name to assign these images:"+rule["name_to_assign"]+NEWLINE)
+          f.write(FOURSPACES+"Name to assign these objects:Cell"+NEWLINE)
+          f.write(FOURSPACES+"Select the image type:"+rule["con_select_image_type"]["select_image_type"]+NEWLINE)
+
+          intensity_range = rule["con_select_image_type"]["con_set_intensity"]["set_intensity_range_from"]
+          f.write(FOURSPACES+"Set intensity range from:"+intensity_range+NEWLINE)
+          f.write(FOURSPACES+"Select the image type:"+rule["con_select_image_type"]["select_image_type"]+NEWLINE)
+
+          if intensity_range=="Manual":
+              f.write(FOURSPACES+"Maximum intensity:"+rule["con_select_image_type"]["con_set_intensity"]["maximum_intensity"]+NEWLINE)
+          else:
+              f.write(FOURSPACES+"Maximum intensity:255.0"+NEWLINE)
+
+          if process_3d =="Yes":
+              f.write(FOURSPACES+"Relative pixel spacing in X:"+x_spacing+NEWLINE)
+              f.write(FOURSPACES+"Relative pixel spacing in Y:"+y_spacing+NEWLINE)
+              f.write(FOURSPACES+"Relative pixel spacing in Z:"+z_spacing+NEWLINE) 
+
+
+
+def write_groups():
+  groups=params['groups']  
+
+  f.write(NEWLINE)
+  
+  f.write("Groups:[module_num:4|svn_version:\\'Unknown\\'|variable_revision_number:2|show_window:False|notes:\\x5B\\\'The Groups module optionally allows you to split your list of images into image subsets (groups) which will be processed independently of each other. Examples of groupings include screening batches, microtiter plates, time-lapse movies, etc.\\'\\x5D|batch_state:array(\\x5B\\x5D, dtype=uint8)|enabled:True|wants_pause:False]"+NEWLINE)
+
+  f.write(FOURSPACES+"Do you want to group your images?:"+groups["con_groups"]["group_images"]+NEWLINE)
+  f.write(FOURSPACES+"grouping metadata count:1"+NEWLINE)
+  f.write(FOURSPACES+"Metadata category:"+groups["con_groups"]["group_category"]+NEWLINE)
+
+
+with open("output","w") as f:
+  headers = ["CellProfiler Pipeline: http://www.cellprofiler.org\n",
+             "Version:3\n",
+             "DateRevision:319\n",
+             "GitHash:\n",
+             "ModuleCount:4\n",
+             "HasImagePlaneDetails:False",
+             "\n"]
+
+  f.writelines(headers)
+
+  write_images()
+  write_metadata()
+  write_nameandtypes()
+  write_groups()
+
+  f.close()
+
+    </configfile>
+  </configfiles>
+  <command><![CDATA[
+    	  python $script_file $inputs

```

Comment:
use single quotes, see also this document: https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices.html

Reply:
added single quotes